aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -185,6 +208,40 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+
+    public entry fun rotate_authentication_key_ed25519(account: &signer, signature: vector<u8>, new_public_key: vector<u8>) acquires Account, OriginatingAddress {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr), error::not_found(EACCOUNT));
+        assert!(
+            vector::length(&new_public_key) == 32,
+            error::invalid_argument(EMALFORMED_PUBLIC_KEY)
+        );
+        assert!(
+            vector::length(&signature) == 64,
+            error::invalid_argument(EMALFORMED_PROOF_OF_KNOWLEDGE)
+        );
+
+        let account_resource = borrow_global_mut<Account>(addr);
+        let current_auth_key = create_address(account_resource.authentication_key);
+
+        let proof = RotationProof {
+            originator: addr,
+            current_auth_key,
+        };
+
+        assert!(signature::ed25519_verify_t(signature, new_public_key, proof), EINVALID_PROOF_OF_KNOWLEDGE);
+        let address_map = &mut borrow_global_mut<OriginatingAddress>(@aptos_framework).address_map;
+        if (table::contains(address_map, current_auth_key)) {
+            table::remove(address_map, current_auth_key);
+        };
+
+        vector::push_back(&mut new_public_key, 0);
```

Comment:
let's put a comment on wtf is happening here lol

(ed25519 authenticator is the sha3 256 hash of the public key with a zero appended)

Reply:
Uhm.... we don't have a function that computes authentication keys? Can we add one?